src/object-store/src/parser/parser.cpp

```diff
@@ -70,37 +70,34 @@ struct argument : seq< one< '$' >, must< argument_index > > {};
 
 // expressions and operators
 struct expr : sor< dq_string, sq_string, number, argument, true_value, false_value, null_value, key_path > {};
-struct case_insensitive : pegtl_istring_t("[c]"){};
+struct case_insensitive : pegtl_istring_t("[c]") {};
 
-struct eq : seq< sor< two< '=' >, one< '=' > >, opt< case_insensitive > >{};
+struct eq : seq< sor< two< '=' >, one< '=' > >, star< blank >, opt< case_insensitive > >{};
```

Comment:
`NSPredicate` allows for spaces, and since we're adopting its syntax, we should too